necromorphs/code/marker/markernet/marker_eye.dm

```diff
@@ -167,11 +183,116 @@ GLOBAL_LIST_EMPTY(markers_signals)
 	interaction_range = null
 	pixel_x = -7
 	pixel_y = -7
+	///Used when spawning necromorphs
+	var/image/necro_preview
+	///Necro class of a necromorph we are going to spawn
+	var/spawning_necromorph
 
 /mob/camera/marker_signal/marker/Initialize(mapload, obj/structure/marker/master)
 	. = ..()
 	icon_state = "mastersignal"
+	verbs -= /mob/camera/marker_signal/verb/become_master
 
 /mob/camera/marker_signal/marker/Destroy()
 	marker?.camera_mob = null
 	return ..()
+
+/mob/camera/marker_signal/marker/verb/downgrade()
+	set name = "Downgrade to normal signal"
+	set category = "Necromorph"
+
+	var/mob/camera/marker_signal/signal = new /mob/camera/marker_signal(get_turf(src), marker)
+	signal.real_name = signal.name
+	signal.ckey = src.ckey
+	qdel(src)
+
+/mob/camera/marker_signal/marker/verb/open_marker_ui()
+	set name = "Open Marker UI"
+	set category = "Necromorph"
+
+	marker.ui_interact(src)
+
+/mob/camera/marker_signal/marker/check_click_intercept(params,A)
+	//Client level intercept
+	if(client?.click_intercept)
+		if(call(client.click_intercept, "InterceptClickOn")(src, params, A))
+			return TRUE
+
+	//Mob level intercept
+	if(click_intercept)
+		if(call(click_intercept, "InterceptClickOn")(src, params, A))
+			return TRUE
+
+	if(spawning_necromorph)
+		var/list/modifiers = params2list(params)
+		if(LAZYACCESS(modifiers, LEFT_CLICK))
+			if(!LAZYACCESS(modifiers, SHIFT_CLICK))
+				spawn_necromorph(A)
+				detach_necro_preview()
+			else
+				spawn_necromorph(A)
+			return TRUE
+		else if(LAZYACCESS(modifiers, RIGHT_CLICK))
+			if(LAZYACCESS(modifiers, SHIFT_CLICK))
+				return FALSE
+			detach_necro_preview()
+
+	return FALSE
+
+/mob/camera/marker_signal/marker/proc/spawn_necromorph(turf/A)
+	if(marker.spent_biomass < marker.necro_classes[spawning_necromorph].biomass_spent_required)
+		to_chat(src, span_warning("You need to spend more biomass to unlock this necromorph!"))
+		return
+	if(marker.biomass < marker.necro_classes[spawning_necromorph].biomass_cost)
+		to_chat(src, span_warning("You don't have enough biomass!"))
+		return
+	A = get_turf(A)
+	if(!A)
+		return
+	if(A.density)
+		to_chat(src, span_warning("Location is dense!"))
+		return
+	for(var/atom/movable/movable as anything in A)
+		if(movable.density)
+			to_chat(src, span_warning("Location has dense objects on it!"))
+			return
+	//In case there was a nearby spawnloc but nest was behind a wall
+	var/spawnloc_cantsee
+	for(var/atom/spawnloc as anything in marker.necro_spawn_atoms)
+		if(get_dist(spawnloc, A) > 4)
```

Comment:
generally recommend `IN_GIVEN_RANGE()` over `get_dist()` due to get dist being quirky.